src/platform-aws.c

```diff
@@ -747,9 +747,9 @@ static int get_rail_vf_idx(struct fi_info *info)
 		NCCL_OFI_WARN("Bad GUID format: wrong colon pos: %s", guid);
 		return -EINVAL;
 	}
-	/* guid[14...] string should now have format ":XXXX". Extract the final two digits
+	/* guid[14...] string should now have format ":XXXX". Extract the final digit
```

Comment:
nit: I'm not a huge fan of the amount of string indexing involved here -- the validation we're doing on the position of the colons or the total length of the string is only a liability, linux doesn't change stuff like that lightly, and validating the output format of a well-known sysfs attribute is unnecessary.

This function doesn't need anything other than `sscanf(..., "%*x:%*x:%*x:%2x", vf_idx);`.

Also, we do want all 8 low bits, right? I don't think the comment above needs to change, we do want the bottom 2 digits. It's definitely worth explicitly commenting that the base is hex, though.

Reply:
yeah, good point.  That's way simpler.